File: packages/sdk-py/examples/basic.py
"""
XASE SDK - Basic Example

Demonstrates fire-and-forget mode for zero latency impact.
"""

import os
import sys

# Add parent directory to path for local development
sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..', 'src'))

from xase import XaseClient

# Initialize client
xase = XaseClient({
    "api_key": os.getenv("XASE_API_KEY", "xase_pk_demo"),
    "base_url": "http://localhost:3000/api/xase/v1",
    "fire_and_forget": True,  # Zero latency impact
    "on_success": lambda result: print(f"✅ Evidence recorded: {result['transaction_id']}"),
    "on_error": lambda error: print(f"❌ Error: {error.code} - {error.message}"),
})


def approve_loan(user_data):
    """Simulate AI agent making a credit decision."""
    print("\n🤖 Processing loan application...")
    
    # Your AI decision logic here
    decision = "APPROVED" if user_data["credit_score"] >= 700 else "DENIED"
    confidence = user_data["credit_score"] / 850
    
    # Record evidence (fire-and-forget, returns immediately)
    xase.record({
        "policy": "credit_policy_v4",
        "input": user_data,
        "output": {"decision": decision},
        "confidence": confidence,
        "transaction_id": f"loan_{user_data['user_id']}_{int(__import__('time').time())}",
        "decision_type": "CREDIT_APPROVAL",
    })
    
    print(f"📝 Decision: {decision} (confidence: {confidence * 100:.1f}%)")
    print("⚡ Evidence queued for async recording (zero latency)")
    
    return decision


def main():
    """Run example."""
    print("🚀 XASE SDK - Basic Example (Python)\n")
    
    # Process multiple loans
    approve_loan({
        "user_id": "u_4829",
        "amount": 50000,
        "credit_score": 720,
    })
    
    approve_loan({
        "user_id": "u_4830",
        "amount": 25000,
        "credit_score": 650,
    })
    
    approve_loan({
        "user_id": "u_4831",
        "amount": 100000,
        "credit_score": 800,
    })
    
    print("\n⏳ Flushing queue before exit...")
    xase.flush(3.0)
    
    print("✅ All evidence recorded!\n")


if __name__ == "__main__":
    main()

File: packages/sdk-js/examples/basic.js
/**
 * XASE SDK - Basic Example
 * 
 * Demonstrates fire-and-forget mode for zero latency impact
 */

const { XaseClient } = require('../dist/index.js')

// Initialize client
const xase = new XaseClient({
  apiKey: process.env.XASE_API_KEY || 'xase_pk_demo',
  baseUrl: 'http://localhost:3000/api/xase/v1',
  fireAndForget: true, // Zero latency impact
  onSuccess: (result) => {
    console.log('✅ Evidence recorded:', result.transaction_id)
  },
  onError: (error) => {
    console.error('❌ Error:', error.code, error.message)
  },
})

// Simulate AI agent making a credit decision
async function approveLoan(userData) {
  console.log('\n🤖 Processing loan application...')
  
  // Your AI decision logic here
  const decision = userData.credit_score >= 700 ? 'APPROVED' : 'DENIED'
  const confidence = userData.credit_score / 850
  
  // Record evidence (fire-and-forget, returns immediately)
  await xase.record({
    policy: 'credit_policy_v4',
    input: userData,
    output: { decision },
    confidence,
    transactionId: `loan_${userData.user_id}_${Date.now()}`,
    decisionType: 'CREDIT_APPROVAL',
  })
  
  console.log(`📝 Decision: ${decision} (confidence: ${(confidence * 100).toFixed(1)}%)`)
  console.log('⚡ Evidence queued for async recording (zero latency)')
  
  return decision
}

// Run example
async function main() {
  console.log('🚀 XASE SDK - Basic Example\n')
  
  // Process multiple loans
  await approveLoan({
    user_id: 'u_4829',
    amount: 50000,
    credit_score: 720,
  })
  
  await approveLoan({
    user_id: 'u_4830',
    amount: 25000,
    credit_score: 650,
  })
  
  await approveLoan({
    user_id: 'u_4831',
    amount: 100000,
    credit_score: 800,
  })
  
  console.log('\n⏳ Flushing queue before exit...')
  await xase.flush(3000)
  
  console.log('✅ All evidence recorded!\n')
}

main().catch(console.error)
